src/biome/data/sources/datasource.py

```diff
@@ -108,79 +105,68 @@ def to_bag(self) -> Bag:
 
         return self._df.to_bag(index=True).map(row2dict, columns=dict_keys)
 
-    def to_forward_bag(self) -> Bag:
-        """Turns the forward DataFrame of the data source into a `dask.Bag` of dictionaries, one dict for each row.
+    def to_mapped_bag(self) -> Bag:
+        """Turns the mapped DataFrame of the data source into a `dask.Bag` of dictionaries, one dict for each row.
         Each dictionary has the column names as keys.
 
         Returns
         -------
         bag
             A `dask.Bag` of dicts.
         """
-        forward_df = self.to_forward_dataframe()
-        dict_keys = [str(column).strip() for column in forward_df.columns]
-        return forward_df.to_bag(index=True).map(row2dict, columns=dict_keys)
+        mapped_df = self.to_mapped_dataframe()
+        dict_keys = [str(column).strip() for column in mapped_df.columns]
+        return mapped_df.to_bag(index=True).map(row2dict, columns=dict_keys)
 
     def to_dataframe(self) -> DataFrame:
-        """Returns the DataFrame of the data source"""
+        """Returns the underlying DataFrame of the data source"""
         return self._df
 
-    def to_forward_dataframe(self) -> DataFrame:
+    def to_mapped_dataframe(self) -> DataFrame:
         """
-        Adds columns to the DataFrame that are named after the parameter names in the model's forward method.
-        The content of these columns is specified in the forward dictionary of the data source yaml file.
+        Adds columns to the DataFrame that are named after the parameter names in the DatasetReader's `text_to_instance`
+        method. The content of these columns is specified in the mapping dictionary.
 
         Returns
         -------
-        forward_dataframe
-            Contains additional columns corresponding to the parameter names of the model's forward method.
+        mapped_dataframe
+            Contains additional columns corresponding to the parameter names
+            of the DatasetReader's `text_to_instance` method.
         """
-        if not self.forward:
-            raise ValueError(
-                "For a 'forward_dataframe' you need to specify a `ForwardConfiguration`!"
-            )
+        if not self.mapping:
```

Comment:
@dcfidalgo I think we can stop validating that. I see use the `to_mapped_dataframe` method as a way to "normalize" the data if required. So there is missing information about mapping we can still apply the normalization, that will do nothing.

What do you think?

Reply:
:+1: